x-pack/plugins/infra/server/services/metrics_explorer_views/metrics_explorer_views_client.ts

```diff
@@ -0,0 +1,216 @@
+
+    return this.mapSavedObjectToMetricsExplorerView(
+      metricsExplorerViewSavedObject,
+      sourceConfiguration.configuration.metricsExplorerDefaultView
+    );
+  }
+
+  public async create(
+    attributes: CreateMetricsExplorerViewAttributesRequestPayload
+  ): Promise<MetricsExplorerView> {
+    this.logger.debug(`Trying to create metrics explorer view ...`);
+
+    // Validate there is not a view with the same name
+    await this.assertNameConflict(attributes.name);
+
+    const metricsExplorerViewSavedObject = await this.savedObjectsClient.create(
+      metricsExplorerViewSavedObjectName,
+      attributes
+    );
+
+    return this.mapSavedObjectToMetricsExplorerView(metricsExplorerViewSavedObject);
+  }
+
+  public async update(
+    metricsExplorerViewId: string,
+    attributes: CreateMetricsExplorerViewAttributesRequestPayload,
+    query: MetricsExplorerViewRequestQuery
+  ): Promise<MetricsExplorerView> {
+    this.logger.debug(
+      `Trying to update metrics explorer view with id "${metricsExplorerViewId}"...`
+    );
+
+    // Validate there is not a view with the same name
+    await this.assertNameConflict(attributes.name, [metricsExplorerViewId]);
+
+    const sourceId = query.sourceId ?? 'default';
+
+    const [sourceConfiguration, metricsExplorerViewSavedObject] = await Promise.all([
+      this.infraSources.getSourceConfiguration(this.savedObjectsClient, sourceId),
+      this.savedObjectsClient.update(
+        metricsExplorerViewSavedObjectName,
+        metricsExplorerViewId,
+        attributes
+      ),
+    ]);
+
+    return this.mapSavedObjectToMetricsExplorerView(
+      metricsExplorerViewSavedObject,
+      sourceConfiguration.configuration.metricsExplorerDefaultView
+    );
+  }
+
+  public delete(metricsExplorerViewId: string): Promise<{}> {
+    this.logger.debug(
+      `Trying to delete metrics explorer view with id ${metricsExplorerViewId} ...`
+    );
+
+    return this.savedObjectsClient.delete(
+      metricsExplorerViewSavedObjectName,
+      metricsExplorerViewId
+    );
+  }
+
+  private mapSavedObjectToMetricsExplorerView(
+    savedObject: SavedObject | SavedObjectsUpdateResponse,
+    defaultViewId?: string
+  ) {
+    const metricsExplorerViewSavedObject = decodeOrThrow(metricsExplorerViewSavedObjectRT)(
+      savedObject
+    );
+
+    return {
+      id: metricsExplorerViewSavedObject.id,
+      version: metricsExplorerViewSavedObject.version,
+      updatedAt: metricsExplorerViewSavedObject.updated_at,
+      attributes: {
+        ...metricsExplorerViewSavedObject.attributes,
+        isDefault: metricsExplorerViewSavedObject.id === defaultViewId,
+        isStatic: false,
+      },
+    };
+  }
+
+  private moveDefaultViewOnTop(views: MetricsExplorerView[]) {
+    const defaultViewPosition = views.findIndex((view) => view.attributes.isDefault);
+
+    if (defaultViewPosition !== -1) {
+      const element = views.splice(defaultViewPosition, 1)[0];
+      views.unshift(element);
```

Comment:
suggestion: We could create a new array instead of mutating the `views` array passed as a parameter.

Reply:
It would make sense for immutability purposes, but as we already modified the original instance with `.splice` to remove the item, there shouldn't be any issue with this mutation.
I measure the performance impact and it's irrelevant in this case.